classify.c

```diff
@@ -376,6 +381,21 @@ static struct irq_info *add_one_irq_to_db(const char *devpath, struct irq_info *
 	return new;
 }
 
+void remove_one_irq_from_db(int irq)
+{
+        struct irq_info find;
+        GList *entry = NULL;
+
+        find.irq = irq;
+        entry = g_list_find_custom(interrupts_db, &find, compare_ints);
+        if (!entry)
+                return;
+
+        interrupts_db = g_list_remove(interrupts_db, entry->data);
```

Comment:
you forgot to free the irq_info struct here (as pointed to by entry->data).  See free_irq_db for a reference to how this is done

Reply:
yeah = =i have fixed it